static/js/components/PersonalTab.js

```diff
@@ -23,15 +39,76 @@ class PersonalTab extends React.Component {
     ui:             UIState,
     nextStep:       () => void,
     prevStep:       () => void,
+    dispatch:       Dispatch,
+    dashboard:      DashboardState,
+    enrollments: ProgramEnrollmentsState,
+    currentProgramEnrollment:    ProgramEnrollment,
   };
 
+  programListing(programs: Program): Array<React$Element<*>> {
+    const makeMenuItems = R.map(program => (
+      <MenuItem value={program.id} key={program.id} primaryText={program.title} />
+    ));
+    const sortPrograms = R.sortBy(R.compose(R.toLower, R.prop('title')));
+    const menuItems = R.compose(makeMenuItems, sortPrograms);
+    return menuItems(programs || {}) ;
+  }
+
+  setProgramHelper(event:Event, key: string, value: string): void {
+    const {
+      dispatch,
+      enrollments: { programEnrollments },
+    } = this.props;
+    let selected = programEnrollments.find(enrollment => enrollment.id === value);
+    dispatch(setCurrentProgramEnrollment(selected));
+  }
+
+  selectProgram(setProgram: Function, program: Program): React$Element<*> {
+    const { dashboard: { programs } } = this.props;
+
+    return (
+      <SelectField
+        style={{width: "65%"}}
+        hintText={program ? program.title : "Select Program"}
+        onChange={setProgram}
+        value={program.id}
+        className="signup-dialog-select"
+      >
+        { this.programListing(programs) }
+      </SelectField>
+    );
+  }
+
   render() {
+    let currentId;
+    const {
+      currentProgramEnrollment,
+      enrollments: {programEnrollments},
+    } = this.props;
+
+    if (!_.isNil(currentProgramEnrollment)) {
+      currentId = currentProgramEnrollment.id;
+    }
+    let selectedProgram = programEnrollments.find(enrollment => enrollment.id === currentId);
+
     return (
       <div>
+        <Card shadow={1}>
+          <div className="profile-form program-select">
+            <div className="section-header">Which MicroMasters program are you signing up for?</div>
+            <br/>
+            { this.selectProgram(this.setProgramHelper, selectedProgram) }
+          </div>
+        </Card>
         <Card shadow={1} className="profile-form">
           <PersonalForm {...this.props} validator={personalValidation} />
           <ValidationAlert {...this.props} />
         </Card>
+        <div className="terms-of-service-container">
+          By clicking "Next" I certify that I agree with <a href="#">
```

Comment:
From https://github.com/mitodl/micromasters/blob/master/static/js/containers/SignupDialog.js#L75-L78

```
/terms_of_service
```

Don't forget to also set `target="_blank"`